backend/expandToppingsCategories.ts

```diff
@@ -0,0 +1,30 @@
+import { getCategoriesCollection } from './db'
+import { Topping, ToppingWithCategoryId } from './topping'
+
+export default async function expandToppingsCategories(
+  toppings: ToppingWithCategoryId[]
+): Promise<Topping[]>
+{
+  const categoriesCollection = await getCategoriesCollection()
+  const categoriesCache: Record<number, string> = {}
+
+  async function getCategoryName(categoryId: number) {
+     if (!(categoryId in categoriesCache)) {
+      const category = await categoriesCollection.findOne({id: categoryId})
+      if (category === null) {
+        throw new Error('Topping category doesn\'t exist')
+      }
+      categoriesCache[categoryId] = category.name
+    }
+    return categoriesCache[categoryId]
+  }
+
+  return Promise.all(
+    toppings.map(
```

Comment:
what is Promise.all ?

Reply:
it converts a list of Promises into a Promise of a list (as .map() doesn't do `await` if you give it an async function)

otherwise I'd need to make a new list using a for loop that awaits each thing

Reply:
Cool 👍